frontends/web/src/routes/lightning/receive/receive.tsx

```diff
@@ -121,87 +122,116 @@ export function Receive({ accounts, code }: Props) {
-              </Button>
-              <Button secondary onClick={back}>
-                {t('button.back')}
-              </Button>
-            </ColumnButtons>
-          </Column>
-        </Grid>
+        <View>
+          <ViewContent>
+            <Grid col="1">
+              <Column>
+                <h1 className={styles.title}>{t('lightning.receive.subtitle')}</h1>
+                <Input
+                  type="number"
+                  min="0"
+                  label={t('lightning.receive.amountSats.label')}
+                  placeholder={t('lightning.receive.amountSats.placeholder')}
+                  id="amountSatsInput"
+                  onInput={onAmountSatsChange}
+                  value={amountSatsText}
+                  autoFocus
+                />
+                <Input
+                  label={t('lightning.receive.description.label')}
+                  placeholder={t('lightning.receive.description.placeholder')}
+                  id="descriptionInput"
+                  onInput={onDescriptionChange}
+                  value={description}
+                  labelSection={
+                    <OptionalLabel>{t('lightning.receive.description.optional')}</OptionalLabel>
+                  }
+                />
+                <Status hidden={!showOpenChannelWarning} type="info">
+                  {t('lightning.receive.openChannelWarning', { feeSat: toSat(openChannelFeeResponse?.feeMsat!) })}
+                </Status>
+              </Column>
+            </Grid>
+          </ViewContent>
+          <ViewButtons>
+            <Button primary onClick={receivePayment} disabled={amountSats === 0}>
+              {t('lightning.receive.invoice.create')}
+            </Button>
+            <Button secondary onClick={back}>
+              {t('button.back')}
+            </Button>
+          </ViewButtons>
+        </View>
       );
     case 'wait':
       return (
-        <Grid col="1">
-          <Column>
-            <QRCode data={receivePaymentResponse?.lnInvoice.bolt11} />
-          </Column>
-          <Column>
-            <ColumnButtons className="m-top-default m-bottom-xlarge" inline>
-              <Button secondary onClick={back} disabled={busy}>
-                {t('button.back')}
-              </Button>
-            </ColumnButtons>
-          </Column>
-        </Grid>
+        <Spinner text="Hardcode invoice creating going…" guideExists={false} />
+      );
+    case 'invoice':
+      return (
+        <View
+          fitContent
+          minHeight="100%">
+          <ViewContent textAlign="center">
+            <Grid col="1">
+              <Column>
+                <h1 className={styles.title}>{t('lightning.receive.invoice.title')}</h1>
+                <div>
+                  <QRCode data={receivePaymentResponse?.lnInvoice.bolt11} />
+                </div>
+                <div className={styles.invoiceSummary}>
+                  {amountSatsText} sats
+                  (--- EUR)
```

Comment:
I imagine this is placeholder for the fiat conversion. FYI I have been tasked to include the conversion rates in the views also later.